app/angular/src/server/angular-cli_config.js

```diff
@@ -21,12 +21,19 @@ export function getAngularCliWebpackConfigOptions(dirToSearch, appIndex = 0) {
     throw new Error('.angular-cli.json must have apps entry.');
   }
   const appConfig = cliConfig.apps[appIndex];
+  const { root, styles, scripts } = appConfig;
```

Comment:
We should read the build options from `angular.json` (not `.angular-cli.json`). And the structures of them are not compatible.

```js
const angularJson = JSON.parse(fs.readFileSync(path.resolve(dirToSearch, 'angular.json'), 'utf8'))
const project = angularJson['projects'][angularJson['defaultProject']];
const cliWebpackConfigOptions = {
  buildOptions: project.architect.build.options,
  root: project.root,
  projectRoot: dirToSearch,
  supportES2015: false,
  tsConfig: {
    options: { },
    fileNames: [],
    errors: [],
  },
  tsConfigPath: path.resolve(dirToSearch, 'src/tsconfig.app.json'),
};
```

I've created the sample repo to extract webpack config object using `@angular-devkit/build-angular` 
https://github.com/Quramy/how-to-extract-webpack-conf-from-angular-cli